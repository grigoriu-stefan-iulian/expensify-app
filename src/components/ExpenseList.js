import React from 'react'
import { connect } from 'react-redux'
import filters from '../reducers/filters';

const ExpenseList = (props) => (
    <div>
        <h1>ExpenseList</h1>
        <p>{props.expenses.length}</p>
        <p>{props.filters.text}</p>
    </div>
)

const mapStateToProps = (state) => ({ 
    expenses: state.expenses,
    filters: state.filters
})

export default connect(mapStateToProps)(ExpenseList)
