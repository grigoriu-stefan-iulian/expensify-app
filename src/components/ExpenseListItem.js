import React from 'react'
import { connect } from 'react-redux'
import { removeExpense } from '../actions/expenses'

const ExpenseListItem = ({ dispatch, description, amount, createdAt, id }) => (
    <div>
        <h4>{description}</h4>
        <p>Amount: {amount}</p>
        <p>Created At: {createdAt}</p>
        <button
            onClick={() => dispatch(removeExpense({ id }))}
        >
            Remove
        </button>
    </div>
)

export default connect()(ExpenseListItem)