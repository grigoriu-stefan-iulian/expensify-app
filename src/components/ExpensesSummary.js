import React from 'react'
import numeral from 'numeral'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({ expensesCount = [], expensesTotal }) => (
    <div>
        <h1>
            Viewing {expensesCount} expense{expensesCount === 1 ?  '' : 's'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}.
        </h1>
    </div>
)

const mapStateToProps = (state) => {
const visibleExpenses = selectExpenses(state.expenses, state.filters)

return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
}
}

export default connect(mapStateToProps)(ExpensesSummary)