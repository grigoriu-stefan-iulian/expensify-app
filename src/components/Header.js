import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <p><NavLink to="/" activeClassName="is-active" exact={true} >Home</NavLink></p>
        <p><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></p>
        <p><NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink></p>
        <p><NavLink to="/help" activeClassName="is-active">Help</NavLink></p>
    </header>
)

export default Header