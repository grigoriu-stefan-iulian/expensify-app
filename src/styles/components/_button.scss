.button {
    color: white;
    border: none;
    font-size: $l-font;
    font-weight: 300;
    padding: $s-size;
}

.button__login-google {
    background: $red;
}

.button__add-expense {
    background: $blue;
}