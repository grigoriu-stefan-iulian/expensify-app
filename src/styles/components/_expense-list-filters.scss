.filters-container {
    display: flex;
    flex-direction: column;
    margin-bottom: $s-size;
    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $m-size;
    }
}

.filters-container__item {
    margin-bottom: $s-size;
    @media (min-width: $desktop-breakpoint) {
        margin: 0 $s-size 0 0;
    }
}