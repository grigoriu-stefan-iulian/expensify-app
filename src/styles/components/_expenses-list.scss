.expenses-list {
    background: $light-grey;
    border: 1px solid darken($light-grey, 10%);
    color: $grey;
    display: flex;
    padding: $m-size;
    justify-content: space-between;
}

.expenses-list__amount {
    margin: $s-size 0 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin: 0 0 0 $m-size;
    }
}

.expenses-list__content {
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}

.expenses-list__item {
    @extend .expenses-list;
    border-top: none;
    color: $off-black;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .2s ease;
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
    }
}

.expenses-list__item:hover {
    background: darken($light-grey, 15%)
}

.expenses-list--no-expenses {
@extend .expenses-list;
}

.expenses-list__sub-title {
    color: $grey;
    font-size: $s-font;
}

.expenses-list__title {
    margin: 0;
    word-break: break-all;
}

