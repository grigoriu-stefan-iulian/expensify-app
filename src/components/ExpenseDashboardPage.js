import React from 'react'
import ConnectedExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        <h2> This is from Dashboard!</h2>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ConnectedExpenseList />
    </div>
)

export default ExpenseDashboardPage