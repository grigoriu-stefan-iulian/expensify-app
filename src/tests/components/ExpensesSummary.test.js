import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render the ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={112.2} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render the ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={21} expensesTotal={11334455662.2}/>)
    expect(wrapper).toMatchSnapshot()
})