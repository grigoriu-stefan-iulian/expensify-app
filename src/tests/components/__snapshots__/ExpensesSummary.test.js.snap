// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totalling 
    $1.12
    .
  </h1>
</div>
`;

exports[`should render the ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    21
     expense
    s
     totalling 
    $113,344,556.62
    .
  </h1>
</div>
`;
